### PR TITLE
LibJS: Implement a couple of normative Temporal changes

### PR DESCRIPTION
**Pull request by Lubrsi**
Implements these two changes:
- https://github.com/tc39/proposal-temporal/commit/6f04074
- https://github.com/tc39/proposal-temporal/commit/374305c

### CODE REVIEW

File: Userland/Libraries/LibJS/Tests/builtins/Temporal/ZonedDateTime/ZonedDateTime.prototype.round.js [hunk 1]
```diff
@@ -112,7 +112,33 @@ describe("errors", () => {
...
+        class CustomTimeZone extends Temporal.TimeZone {
+            constructor() {
+                super("UTC");
+                this.getPossibleInstantsForCallCount = 0;
+            }
+
+            getPossibleInstantsFor(temporalDateTime) {
+                this.getPossibleInstantsForCallCount++;
+
+                if (this.getPossibleInstantsForCallCount === 2) return [new Temporal.Instant(-1n)];
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Temporal/ZonedDateTime/ZonedDateTime.prototype.round.js:129]: Please insert curlies here, the prettier formatting is hideous

**Review by linusg — COMMENTED** [review]
Breaks test-js on CI

**Comment by Lubrsi** [comment]
> Breaks test-js on CI

Ah, I forgot to re-run it after splitting up the error message.

**Review by linusg — APPROVED** [review]
Thanks! :^)

**Pull request merged**